src/server/hapi.ts

```diff
@@ -20,6 +20,8 @@ declare module 'hapi' {
   interface RequestDecorators {
     userHasAccessToProject: (projectId: number) => Promise<boolean>;
     userHasAccessToTeam: (teamId: number) => Promise<boolean>;
+    isOpenDeployment: (projectId: number, deploymentId: number) => Promise<boolean>;
+    getOpenTeamId: () => Promise<number | undefined>;
```

Comment:
I'd maybe get rid of this function altogether. More info in the main review comment.